config/src/main/java/com/typesafe/config/impl/SimpleConfig.java

```diff
@@ -282,20 +283,20 @@ public Object getAnyRef(String path) {
 
     @Override
     public Long getBytes(String path) {
-        Long size = null;
-        try {
-            size = getLong(path);
-        } catch (ConfigException.WrongType e) {
-            ConfigValue v = find(path, ConfigValueType.STRING);
-            size = parseBytes((String) v.unwrapped(),
-                    v.origin(), path);
-        }
-        return size;
+        return getMemorySize(path).toBytes();
```

Comment:
Here I think we'd now throw IllegalArgumentException when we used to throw ConfigException.BadValue ... that seems like an API break that could mess up somebody's code. Possible solution: a variant of ConfigMemorySize.toBytes that takes the origin/path information. Another option: call toBytesBigInteger here then do the range check here.

Reply:
sorry, my fault, fixed, thank you.